### PR TITLE
Fix: fix a bug related to navigation to function in text mode

### PR DESCRIPTION
**Pull request by YannanGao-gs**
## Summary

Fixes https://github.com/finos/legend-studio/issues/1848
## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

https://user-images.githubusercontent.com/73408381/214389878-65e25470-1fd9-4ea3-bc1d-b5f0c8f61792.mov

### CODE REVIEW

**Review by xannem — APPROVED** [review]

File: packages/legend-application-studio/src/stores/editor-state/GrammarTextEditorState.ts [hunk 1]
```diff
@@ -183,7 +184,14 @@ export class GrammarTextEditorState {
     }
     this.currentElementLabelRegexString = getGrammarElementTypeLabelRegexString(
       typeLabel,
-      element.path,
+      `${element.package?.path}::${generatePackageableElementTreeNodeDataLabel(
+        element,
+      )
+        .replaceAll('*', '\\*')
```
Comment (MauricioUyaguari) [inline packages/legend-application-studio/src/stores/editor-state/GrammarTextEditorState.ts:190]: what are we trying to replace here?

Comment (YannanGao-gs) [reply packages/legend-application-studio/src/stores/editor-state/GrammarTextEditorState.ts:190]: make sure we could identify *, [, ], (, )  when using regex

Comment (MauricioUyaguari) [reply packages/legend-application-studio/src/stores/editor-state/GrammarTextEditorState.ts:190]: shouldnt the logic then live in `getGrammarElementTypeLabelRegexString`

**Comment by MauricioUyaguari** [comment]
Test for this ?

**Comment by YannanGao-gs** [comment]
> Test for this ?

one test has been added.

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**